weed/s3api/s3api_object_handlers.go

```diff
@@ -35,10 +37,17 @@ func urlEscapeObject(object string) string {
 	return "/" + t
 }
 
+func entryUrlEncode(dir string, entry string, encodingTypeUrl bool) (dirName string, entryName string, prefix string) {
+	if !encodingTypeUrl {
+		return dir, entry, entry
+	}
+	return urlPathEscape(dir), url.QueryEscape(entry), urlPathEscape(entry)
+}
+
 func urlPathEscape(object string) string {
 	var escapedParts []string
 	for _, part := range strings.Split(object, "/") {
-		escapedParts = append(escapedParts, url.PathEscape(part))
+		escapedParts = append(escapedParts, strings.ReplaceAll(url.PathEscape(part), "+", "%2B"))
```

Comment:
this `strings.ReplaceAll` seems only one specific escaping. Would it break if other characters need to be replaced.

Reply:
Otherwise, nothing else needs to be replaced, and for some reason there was no ready-made function, since this is some kind of aws specific encodin.